android/src/main/java/org/conscrypt/Platform.java

```diff
@@ -184,35 +170,16 @@ public static void setSocketWriteTimeout(Socket s, long timeoutMillis) throws So
             }
 
             Field f_SOL_SOCKET = c_osConstants.getField("SOL_SOCKET");
-            if (f_SOL_SOCKET == null) {
-                Log.w(TAG, "SOL_SOCKET == null; not setting socket write timeout");
-                return;
-            }
-
             Field f_SO_SNDTIMEO = c_osConstants.getField("SO_SNDTIMEO");
-            if (f_SO_SNDTIMEO == null) {
-                Log.w(TAG, "SO_SNDTIMEO == null; not setting socket write timeout");
-                return;
-            }
-
             Method m_setsockoptTimeval = instance_os.getClass().getMethod("setsockoptTimeval",
                     FileDescriptor.class, int.class, int.class, c_structTimeval);
-            if (m_setsockoptTimeval == null) {
-                Log.w(TAG, "setsockoptTimeval == null; not setting socket write timeout");
-                return;
-            }
 
             m_setsockoptTimeval.invoke(instance_os, fd, f_SOL_SOCKET.get(null),
                     f_SO_SNDTIMEO.get(null), timeval);
+        } catch (SocketException e) {
+            throw e;
         } catch (Exception e) {
-            // We don't want to spam the logcat since this isn't a fatal error, but we want to know
-            // why this might be happening.
-            logStackTraceSnippet("Could not set socket write timeout: " + e, e);
-            Throwable cause = e.getCause();
-            while (cause != null) {
-                logStackTraceSnippet("Caused by: " + cause, cause);
-                cause = cause.getCause();
-            }
+            // All reflection errors silently ignored.
```

Comment:
nit: wouldn't it be safer towards future changes to list these errors instead of catching everything? 

Should we log it as debug?